class io.netty.channel.CombinedChannelDuplexHandler$1 extends io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext
{
final io.netty.channel.CombinedChannelDuplexHandler this$0;
void <init>(io.netty.channel.CombinedChannelDuplexHandler, io.netty.channel.ChannelHandlerContext, io.netty.channel.ChannelHandler)
{
io.netty.channel.ChannelHandlerContext v2;
io.netty.channel.CombinedChannelDuplexHandler v1;
io.netty.channel.ChannelHandler v3;
io.netty.channel.CombinedChannelDuplexHandler$1 v0;
v0 := @this: io.netty.channel.CombinedChannelDuplexHandler$1;
v1 := @parameter0: io.netty.channel.CombinedChannelDuplexHandler;
v2 := @parameter1: io.netty.channel.ChannelHandlerContext;
v3 := @parameter2: io.netty.channel.ChannelHandler;
v0.<io.netty.channel.CombinedChannelDuplexHandler$1: io.netty.channel.CombinedChannelDuplexHandler this$0> = v1;
specialinvoke v0.<io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext: void <init>(io.netty.channel.ChannelHandlerContext,io.netty.channel.ChannelHandler)>(v2, v3);
return;
}
public io.netty.channel.ChannelHandlerContext fireExceptionCaught(java.lang.Throwable)
{
java.lang.Throwable v1, v9;
java.lang.String v13;
boolean v11, v15, v4;
io.netty.channel.CombinedChannelDuplexHandler v2, v5, v7;
io.netty.channel.ChannelOutboundHandler v6;
io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext v3, v8;
io.netty.util.internal.logging.InternalLogger v10, v12, v14, v16;
io.netty.channel.CombinedChannelDuplexHandler$1 v0;
v0 := @this: io.netty.channel.CombinedChannelDuplexHandler$1;
v1 := @parameter0: java.lang.Throwable;
v2 = v0.<io.netty.channel.CombinedChannelDuplexHandler$1: io.netty.channel.CombinedChannelDuplexHandler this$0>;
v3 = v2.<io.netty.channel.CombinedChannelDuplexHandler: io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext outboundCtx>;
v4 = v3.<io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext: boolean removed>;
if v4 != 0 goto label5;
label1:
v5 = v0.<io.netty.channel.CombinedChannelDuplexHandler$1: io.netty.channel.CombinedChannelDuplexHandler this$0>;
v6 = v5.<io.netty.channel.CombinedChannelDuplexHandler: io.netty.channel.ChannelOutboundHandler outboundHandler>;
v7 = v0.<io.netty.channel.CombinedChannelDuplexHandler$1: io.netty.channel.CombinedChannelDuplexHandler this$0>;
v8 = v7.<io.netty.channel.CombinedChannelDuplexHandler: io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext outboundCtx>;
interfaceinvoke v6.<io.netty.channel.ChannelOutboundHandler: void exceptionCaught(io.netty.channel.ChannelHandlerContext,java.lang.Throwable)>(v8, v1);
label2:
goto label6;
label3:
v9 := @caughtexception;
v10 = <io.netty.channel.CombinedChannelDuplexHandler: io.netty.util.internal.logging.InternalLogger logger>;
v11 = interfaceinvoke v10.<io.netty.util.internal.logging.InternalLogger: boolean isDebugEnabled()>();
if v11 == 0 goto label4;
v12 = <io.netty.channel.CombinedChannelDuplexHandler: io.netty.util.internal.logging.InternalLogger logger>;
v13 = staticinvoke <io.netty.util.internal.ThrowableUtil: java.lang.String stackTraceToString(java.lang.Throwable)>(v9);
interfaceinvoke v12.<io.netty.util.internal.logging.InternalLogger: void debug(java.lang.String,java.lang.Object,java.lang.Object)>("An exception {}was thrown by a user handler\'s exceptionCaught() method while handling the following exception:", v13, v1);
goto label6;
label4:
v14 = <io.netty.channel.CombinedChannelDuplexHandler: io.netty.util.internal.logging.InternalLogger logger>;
v15 = interfaceinvoke v14.<io.netty.util.internal.logging.InternalLogger: boolean isWarnEnabled()>();
if v15 == 0 goto label6;
v16 = <io.netty.channel.CombinedChannelDuplexHandler: io.netty.util.internal.logging.InternalLogger logger>;
interfaceinvoke v16.<io.netty.util.internal.logging.InternalLogger: void warn(java.lang.String,java.lang.Object,java.lang.Object)>("An exception \'{}\' [enable DEBUG level for full stacktrace] was thrown by a user handler\'s exceptionCaught() method while handling the following exception:", v9, v1);
goto label6;
label5:
specialinvoke v0.<io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext: io.netty.channel.ChannelHandlerContext fireExceptionCaught(java.lang.Throwable)>(v1);
label6:
return v0;
catch java.lang.Throwable from label1 to label2 with label3;
}
}